src/duke/Storage.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import java.io.*;
+import java.util.ArrayList;
+
+
+/**
+ * This is an FileAction class that allows the saving of
+ * contents into a pre-specified file
+ *
+ * @author  Hsiao Jiet
+ * @version 1.0
+ * @since   2022-1-25
+ */
+
+public class Storage {
+    protected String filepath;
+    protected FileWriter fileWriter;
+    protected FileReader fileReader;
+    protected BufferedWriter writer;
+    protected BufferedReader reader;
+    protected TaskList tasks;
+
+    public Storage(String path, TaskList tasks) {
+        filepath = path;
+        this.tasks = tasks;
+    }
+
+    /**
+     * Saves content of Tasks into given file
+     */
+    public void saveTask(String cont) throws IOException {
+        try {
+            writer.write(cont);
+        } catch (IOException e) {
+            throw new IOException("\nUnexpected error occurred where Tasks cannot be saved in file!\n");
+        }
+    }
+
+    /**
+     * Starts the process of file writing
+     */
+    public void startWriter() throws IOException {
+        fileWriter = new FileWriter(filepath);
+        writer = new BufferedWriter(fileWriter);
+    }
+
+    /**
+     * Asks for user input to change file read/to be written to what user inputted
+     */
+    public void requestChangeFile(String fileName) {
+        filepath = fileName;
+    }
+
+    /**
+     * Closes file after writing into the given file
+     */
+    public void closeWriteFile() throws IOException {
+        try {
+            writer.close();
+        } catch (IOException e) {
+            throw new IOException("\nUnexpected error occurred where file writer cannot be closed!\n");
+        }
+    }
+
+    /**
+     * Closes file after reading the given file
+     */
+    public void closeReadFile() throws IOException {
+        try {
+            reader.close();
+        } catch (IOException e) {
+            throw new IOException("\nUnexpected error occurred where file reader cannot be closed!\n");
+        }
+    }
+
+    /**
+     * Saves the current instances of duke.Task items in Tasks list into
+     * provided text file
+     */
+    public void saveAllTasks(TaskList tasks) throws IOException {
+        startWriter();
+        for (Task task : tasks.getTasks()) {
+            saveTask(task.toString());
+        }
+    }
+
+    /**
+     * Reads tasks saved in the hard drive file provided by the user and
+     * places duke.Task objects into the current DukeList
+     */
+    public void readTasksFromFile() throws FileNotFoundException {
+        ArrayList<Task> readTasks = new ArrayList<>();
```

Comment:
Should `readTasks` be a noun? like `tasksThatHaveBeenRead`

Reply:
In this case, read can mean the past-tense version of it, like "I read a book last week", and not "I read textbooks for fun"

Reply:
Ah you're right, this can be quite confusing for people who do not know how I code. Thanks a lot!

Reply:
I see, sorry for my bad English haha 🤣

Reply:
No no, I am saying that the way I named the function wasn't informational enough for users (confusing to them), that's why thanks for the recommendation! LOL